src/Stripe.net/Services/Invoices/StripeInvoiceService.cs

```diff
@@ -131,6 +136,15 @@ public virtual async Task<StripeInvoice> PayAsync(string invoiceId, StripeReques
                     cancellationToken).ConfigureAwait(false));
         }
 
+        public virtual async Task<StripeInvoice> PayAsync(string invoiceId, StripeInvoicePayOptions payOptions, StripeRequestOptions requestOptions = null, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
It seems you an not call PayAsync with the option from the one without, it returns a compilation error

```

Services/Invoices/StripeInvoiceService.cs(132,20): error CS4016: Since this is an async method, the return expression must be of type 'StripeInvoice' rather than 'Task<StripeInvoice>' [/Users/remi/Workspace/bindings/stripe-dotnet/src/Stripe.net/Stripe.net.csproj]
```

Reply:
You want `return await PayAsync(...)`;

Reply:
damn I tried a few variations but did not think it just wanted the `await`. Thanks!